docs/enhancements/1-crds-and-kubebuilder.md

```diff
@@ -0,0 +1,380 @@
+        ).
+        Complete(r)
+}
+```
+
+All export-related resource will contain ownerReference pointing to `FederatedServicePolicy`,
+so deleting `FederatedServicePolicy` will result in removing these resources.
+```yaml
+apiVersion: networking.istio.io/v1
+kind: Gateway
+metadata:
+  name: federation-ingress-gateway
+  namespace: istio-system
+  ownerReferences:
+  - apiVersion: federation.openshift-service-mesh.io/v1alpha1
+    kind: FederatedServicePolicy
+    name: default
+    uid: a8e825b9-911e-40b8-abff-58f37bb3e05d
+spec:
+  selector:
+    app: federation-ingress-gateway
+```
+
+#### ImportedService
+
+`ImportedService` will be created for or updated by the FDS client for each imported service.
+A dedicated controller will watch this CR and manage its child resources, like `ServiceEntry`, `WorkloadEntry`, `DestinationRule`, etc.
+Its owner will be `MeshFederation`, so when `MeshFederation` will be removed `ImportedService` and its child resources will be removed.
+
+```yaml
+apiGroup: federation.openshift-service-mesh.io/v1alpha1
+kind: ImportedService
+metadata:
+  name: <generated-name>
+  namespace: istio-system
+  ownerReferences:
+  - apiVersion: federation.openshift-service-mesh.io/v1alpha1
+    kind: MeshFederation
+    name: default
+    uid: a8e825b9-911e-40b8-abff-58f37bb3e05d
+spec:
+  # importAsLocal specifies if the controller needs to create ServiceEntry or WorkloadEntry and what namespace should be
+  # used for child resources - root mesh namespace or the original namespace.
+  importAsLocal: false
+  name: <original-svc-name>
+  namespace: <original-svc-namespace>
+# TODO
+# status:
+```
+
+Example child resource
+```yaml
+apiVersion: networking.istio.io/v1
+kind: ServiceEntry
+metadata:
+  name: import-productpage-bookinfo
+  namespace: istio-system
+  ownerReferences:
+  - apiVersion: federation.openshift-service-mesh.io/v1alpha1
+    kind: ImportedService
+    name: productpage
+    uid: a8e825b9-911e-40b8-abff-58f37bb3e05d
+spec:
+  hosts:
+  - productpage.bookinfo.svc.cluster.local
+  ...
+```
+
+A controller for `ImportedService` will look like:
+```go
+func (r *ImportedServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+    return ctrl.NewControllerManagedBy(mgr).
+        For(&v1alpha1.ImportedService{}).
+        Owns(&v1.ServiceEntry{}).
+        Owns(&v1.WorkloadEntry{}).
+        Owns(&v1.DestinationRule{}).
+        Watches(
+            &corev1.Service{},
+            handler.EnqueueRequestsFromMapFunc(),
+            builder.WithPredicates(checkIfMatchesExportRules),
+        ).
+        Watches(
+            &corev1.Namespace{},
+            handler.EnqueueRequestsFromMapFunc(),
+            builder.WithPredicates(checkIfMatchesExportRules),
+        ).
+        Watches(
+            &v1alpha1.MeshFederation{},
+            handler.EnqueueRequestsFromMapFunc(),
+        ).
+        Complete(r)
+}
+```
+
+`ImportedService` will be applied by FDS client like below:
```

Comment:
I think we could rely on `EventRecorder` to trigger reconciliation without the actual event in k8s, instead of applying objects. I will investigate this approach.

Reply:
Ok, I was looking for a workaround to trigger reconciliation without creating objects in k8s. We can try to trigger reconciliation using a channel.

Reply:
*I mentioned this on slack a while back, but I think it's worth adding it here too:* If we want to trigger reconcile from some external flow (i.e. not a k8s event) we can use [`source.Channel`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/source#Channel) for that.